docs/src/main/java/io/helidon/docs/se/inject/FactoryExample.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.docs.se.inject;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import io.helidon.common.GenericType;
+import io.helidon.service.registry.Lookup;
+import io.helidon.service.registry.Qualifier;
+import io.helidon.service.registry.Service;
+
+class FactoryExample {
+
+    record MyService() {}
+
+    // tag::snippet_1[]
+    /**
+     * Supplier service factory.
+     */
+    @Service.Singleton
+    class MyServiceProvider implements Supplier<MyService> {
+
+        @Override
+        public MyService get() {
+            return new MyService();
+        }
+    }
+    // end::snippet_1[]
+
+    // tag::snippet_2[]
+    @Service.Singleton
+    class MyServiceFactory implements Service.ServicesFactory<MyService> {
+        @Override
+        public List<Service.QualifiedInstance<MyService>> services() {
+            return List.of(Service.QualifiedInstance.create(new MyService()));
+        }
+    }
+    // end::snippet_2[]
+
+    // tag::snippet_3[]
+    @Service.Singleton
+    @Service.Named("name")
+    class MyServiceFactoryWithQualifier implements Service.ServicesFactory<MyService> {
+        @Override
+        public List<Service.QualifiedInstance<MyService>> services() {
+            return List.of(Service.QualifiedInstance.create(new MyService(), Qualifier.createNamed("name")));
+        }
+    }
+    // end::snippet_3[]
+
+    // tag::snippet_4[]
+    @Service.Singleton
+    class QualifiedFactory implements Service.QualifiedFactory<MyService, Service.Named> {
```

Comment:
Please never use `Named` in examples that are not showing how named works.
Use a custom qualifier (i.e. `@ConfigProperty` or something like that) for examples showing qualified factories (and anything else related to qualifiers).
Named has a slightly different handling.

Reply:
Changed